File: Edu-Python/src/com/jetbrains/edu/python/learning/checkio/newProject/PyCheckiOCourseProjectGenerator.java
package com.jetbrains.edu.python.learning.checkio.newProject;

import com.jetbrains.edu.learning.courseFormat.Course;
import com.jetbrains.edu.python.learning.PyCourseBuilder;
import com.jetbrains.edu.python.learning.newproject.PyCourseProjectGenerator;
import org.jetbrains.annotations.NotNull;

public class PyCheckiOCourseProjectGenerator extends PyCourseProjectGenerator {
  public PyCheckiOCourseProjectGenerator(@NotNull PyCourseBuilder builder,
                                         @NotNull Course course) {
    super(builder, course);
  }

  @Override
  protected boolean beforeProjectGenerated() {
    return true;
  }
}

File: Edu-Python/src/com/jetbrains/edu/python/learning/checkio/pycharm/PyCheckiOCourseProjectGenerator.kt
package com.jetbrains.edu.python.learning.checkio.pycharm

import com.intellij.openapi.project.Project
import com.jetbrains.edu.learning.courseFormat.Course
import com.jetbrains.edu.python.learning.PyCourseBuilder
import com.jetbrains.edu.python.learning.checkio.PyCheckiOCourseProjectGenerator
import com.jetbrains.edu.python.learning.pycharm.PyCourseProjectGenerator
import com.jetbrains.python.newProject.PyNewProjectSettings

internal class PyCheckiOCourseProjectGenerator(builder: PyCourseBuilder, course: Course) : PyCheckiOCourseProjectGenerator(builder, course) {
  private val myPyCourseProjectGenerator = PyCourseProjectGenerator(builder, course)

  override fun afterProjectGenerated(project: Project, settings: PyNewProjectSettings) =
    myPyCourseProjectGenerator.afterProjectGenerated(project, settings)
}
